Client/Frontend/Toolbar+URLBar/TabLocationView.swift

```diff
@@ -328,7 +314,9 @@ private extension TabLocationView {
     func setReaderModeState(_ newReaderModeState: ReaderModeState) {
         let wasHidden = readerModeButton.isHidden
         self.readerModeButton.readerModeState = newReaderModeState
-        readerModeButton.isHidden = (newReaderModeState == ReaderModeState.unavailable) || isShoppingCartButtonVisible
+        let product = url.flatMap(ShoppingProduct.init)
+        let isShoppingCartButtonVisible = product?.isShoppingCartButtonVisible ?? false
```

Comment:
Alternatively we could save the product when the url is set and use it here.
 ```suggestion

            var isShoppingCartButtonVisible = false
            if let url = url {
                let product = ShoppingProduct(url: url)
                isShoppingCartButtonVisible = product.isShoppingCartButtonVisible
            }
            
```